plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/internal/crc/server/core/adapter/CRC100Poller.java

```diff
@@ -153,8 +153,28 @@ protected IStatus runOnePingJson(IServer server, Map<String, String> env)
 				String allContent = String.join("\n", lines);
 			    ObjectMapper mapper = new ObjectMapper();
 			    JsonNode actualObj = mapper.readTree(allContent);
-			    JsonNode field = actualObj.get("crcStatus");
-			    String status = field.asText();
+				// Since crc 1.22.0 there is another patch that breaks this code.
+				// https://github.com/code-ready/crc/issues/1909
+				// Now "./crc status --output json" output does not end on stdErr
+				// but returns stdOut and json string which we have to handle.
+			    if (!actualObj.has("success") && !actualObj.has("crcStatus")) {
+			    	// not really sure about this exception and the state when it might occur
+			    	throw new PollingException("cannot derive crc status, json retrieved: \n\r" + actualObj.toPrettyString());
+			    }
+			    JsonNode successField = actualObj.get("success");
+			    String success = successField.asText();
+			    if (success.equals("false")) {
+			    	if (actualObj.has("error")) {
+					    JsonNode errorField = actualObj.get("error");
+					    String error = errorField.asText();
+				    	if (error.contains("Machine does not exist")) {
```

Comment:
What is the error message is different ? We may endup with the same NPE as before

Reply:
Error message did not change. But the `crc status -o json` did. Before it sent this kind of message to the stdErr. Now it is on stdOut. There are other options as of what can error could possibly contain, so I have to check that it not yet started crc like "error". 

There are multiple options that could fill in error field: https://github.com/code-ready/crc/blob/f51090d9f0086b822daf41503a16978d826c0b67/cmd/crc/cmd/status.go#L49.

Reply:
Why don't you simply trust the success=false flag so that we will be error free when crc decided to change their error message

Reply:
Because I think that simply trusting success=false does not always indicate that machine is in "not yet started, but otherwise alright" state. I believe that there are some situations where you should not let adapter into "starting" state, but this needs more details from crc side. If it would be addressed in sprint to come I am ok with that.